Yosemite/YosemiteTests/Stores/ProductVariationStoreTests.swift

```diff
@@ -289,6 +289,88 @@ final class ProductVariationStoreTests: XCTestCase {
         store.onAction(action)
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
+
+    // MARK: - ProductVariationAction.updateProductVariation
+
+    /// Verifies that `ProductVariationAction.updateProductVariation` returns the expected `ProductVariation`.
+    ///
+    func testUpdatingProductVariationReturnsExpectedFieldsAndRelatedObjects() {
+        // Given
+        let remote = MockProductVariationsRemote()
+        let store = ProductVariationStore(dispatcher: dispatcher, storageManager: storageManager, network: network, remote: remote)
+        let productVariationID: Int64 = 17
+        let expectedProductVariation = MockProductVariation()
+            .productVariation(siteID: sampleSiteID, productID: sampleProductID, variationID: productVariationID)
+            .copy(attributes: sampleProductVariationAttributes())
+        let productVariation = MockProductVariation().productVariation(siteID: sampleSiteID, productID: sampleProductID, variationID: productVariationID)
+            .copy(description: "Wooooo", sku: "test-woo")
+        remote.whenUpdatingProduct(siteID: sampleSiteID,
+                                   productID: sampleProductID,
+                                   productVariationID: productVariationID,
+                                   thenReturn: .success(expectedProductVariation))
+
+        // Saves an existing ProductVariation into storage.
+        // Note: at least one field of `ProductVariation` before and after the update should be different.
+        storageManager.insertSampleProductVariation(readOnlyProductVariation: productVariation)
+        XCTAssertEqual(viewStorage.countObjects(ofType: StorageProductVariation.self), 1)
+
+        // When
+        var result: Result<Yosemite.ProductVariation, ProductUpdateError>?
+        waitForExpectation { expectation in
+            let action = ProductVariationAction.updateProductVariation(productVariation: productVariation) { aResult in
+                result = aResult
+                expectation.fulfill()
+            }
+            store.onAction(action)
+        }
+
+        // Then
+        guard case let .success(updatedProductVariation) = result else {
+            XCTFail("Unexpected result: \(String(describing: result))")
+            return
+        }
```

Comment:
We can also do this:

```suggestion
        let updatedProductVariation = try result?.get()
```

Reply:
yup! updated in https://github.com/woocommerce/woocommerce-ios/pull/2539/commits/0a372c05c4034812cefeb3ef6395e2d3df225cd5